### PR TITLE
Remote ActorRef Provider Fix for #1646

### PR DESCRIPTION
**Pull request by rogeralsing**
Fix for #1646

In the `RemoteDeadletterActorRef` there is some missing code to propagate deadletters to the base implementation:

```
.With<DeadLetter>(
    deadLetter =>
    {
        // else ignore: it is a reliably delivered message that might be retried later, and it has not yet deserved
        // the dead letter status
        var deadSend = deadLetter.Message as EndpointManager.Send;
        if (deadSend != null && deadSend.Seq == null)
        {
            base.TellInternal(deadSend.Message, deadSend.SenderOption ?? ActorRefs.NoSender);
        }
       
        //##############
       // the issue here was that the With<DeadLetter> consumed _all_ dead letters, 
       // when this clause was just intended to consume those containing EndpointManager.Send
    })
```

### CODE REVIEW

File: src/core/Akka.Remote/RemoteActorRefProvider.cs [hunk 1]
```diff
@@ -532,30 +532,27 @@ public RemoteDeadLetterActorRef(IActorRefProvider provider, ActorPath actorPath,
...
+                var send = message as EndpointManager.Send;
+                var deadLetter = message as DeadLetter;
+                if (send != null)
+                {
+                    if (send.Seq == null)
+                    {
+                        base.TellInternal(send.Message, send.SenderOption ?? ActorRefs.NoSender);
+                    }
+                }
+                else if (deadLetter?.Message is EndpointManager.Send)
```
Comment (rogeralsing) [inline src/core/Akka.Remote/RemoteActorRefProvider.cs]: I've replaced the pattern matching code with normal if blocks, this specific check only consumes deadletters containing EndpointManager.Send messages now.
(unlike before where it incorrectly consumed all deadletters)

**Comment by rogeralsing** [comment]
See Scala code for ref implementation

```scala
override def !(message: Any)(implicit sender: ActorRef): Unit = message match {
      case Send(m, senderOption, _, seqOpt) ⇒
        // else ignore: it is a reliably delivered message that might be retried later, and it has not yet deserved
        // the dead letter status
        if (seqOpt.isEmpty) super.!(m)(senderOption.orNull)
      case DeadLetter(Send(m, senderOption, recipient, seqOpt), _, _) ⇒
        // else ignore: it is a reliably delivered message that might be retried later, and it has not yet deserved
        // the dead letter status
        if (seqOpt.isEmpty) super.!(m)(senderOption.orNull)
      case _ ⇒ super.!(message)(sender)
    }
```

**Comment by Aaronontheweb** [comment]
I've reviewed this and yep, the bug is pretty clearly a greedy match statement and @rogeralsing's fix matches the JVM source. Merging it in prior to 1.0.6.

**Comment by Aaronontheweb** [comment]
Crap, hit the wrong button the first time

**Pull request merged**